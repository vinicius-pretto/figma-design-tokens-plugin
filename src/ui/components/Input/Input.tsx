import * as React from "react";

interface Props {
  id: string;
  value: string;
  onChange: any;
  placeholder?: string;
}

const Input = ({ id, value, onChange, placeholder }: Props) => {
  return (
    <div className="input">
      <input
        id={id}
        className="input__field mb-md"
        value={value}
        onChange={onChange}
        placeholder={placeholder}
      />
    </div>
  );
};

export default Input;
