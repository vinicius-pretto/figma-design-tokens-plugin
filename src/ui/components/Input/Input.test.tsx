import * as React from "react";
import { render } from "@testing-library/react";
import Input from "./Input";

describe("Input", () => {
  it("snapshot", () => {
    const { asFragment } = render(
      <Input
        id="name"
        type="text"
        value="#cc0000"
        onChange={jest.fn()}
        placeholder="color-primary"
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
