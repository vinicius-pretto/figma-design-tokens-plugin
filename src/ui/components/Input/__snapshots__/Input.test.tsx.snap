// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input snapshot 1`] = `
<DocumentFragment>
  <div
    class="input"
  >
    <input
      class="input__field mb-md"
      id="name"
      placeholder="color-primary"
      value="#cc0000"
    />
  </div>
</DocumentFragment>
`;
